tf_quant_finance/examples/jupyter_notebooks/Swap_Curve_Fitting.ipynb

```diff
@@ -0,0 +1,533 @@
+      "text/plain": [
+       "<Figure size 864x576 with 1 Axes>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "plt.plot(curve.times, curve.rates, label='Discount rates')\n",
+    "plt.legend()\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Fit the interest rate curve"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The data is retrieved from Bloomberg on 2020-02-14 to fit the USD LIBOR 3M curve. \n",
+    "\n",
+    "The following market instruments are used\n",
+    "\n",
+    "- LIBOR Fixed 3 Month: Spot LIBOR 3M rate\n",
+    "\n",
+    "- ED Futures: Eurodollar Futures with expiry up to 2Y\n",
+    "\n",
+    "- Interest Rate Swap: Liquid interest rate swap instruments with tenors from 2 years to 50 years\n",
+    "\n",
+    "The basis of the floating rate is ACT/360. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "today = datetime(2020, 2, 14)\n",
```

Comment:
Could you please use `dates` module from tff?
```
 dates = tff.experimental.dates
today = dates.from_datetimes([datetime(2020, 2, 14)])
```
The reason is that the operations are tensorized, meaning you can do things like this:
```
start_dates = dates.from_datetimes([datetime(2020, 3, 16), datetime(2020, 7, 13)])  # put all your dates here
daycount = dates.daycounts.actual_360  # we have daycounting now
daycount(start_date=today, end_date=start_dates)
# Outputs [0.08611111, 0.41666666]
```
As for the current example, I would suggest rewriting it as follows. Let the input data be a dictionary of what you have as the arguments. E.g., cash_intrument is characterized by (tenor, rate), so the input is, say, a dictionary of  the form
{'cash_instrument_id': {'tenor', 'rate'}} (this would mean tcorrespond to the case when the user can stores the data as json). Keep all the dates as actual dates. For the swaps, create schedules using `dates.PeriodicSchedule`. Also, you can add dates, if you find that useful: `today + dates.periods.years(3)`. 

This way we can get rid of numpy and use tensorflow in most places.

Reply:
Still learning to use them and updated in c537ef8b7d6946faeb832c2a90a18d0eab5d38f0.